sbi/inference/posteriors/base_posterior.py

```diff
@@ -607,3 +737,83 @@ def __str__(self):
         )
 
         return desc
+
+
+class ConditionalPotentialFunctionProvider:
+    """
+    Wraps the potential functions to allow for sampling from the conditional posterior.
+    """
+
+    def __init__(
+        self,
+        potential_fn_provider: Callable,
+        condition: Tensor,
+        dims_to_sample: List[int],
+    ):
+        """
+        Args:
+            potential_fn_provider: Creates potential function of unconditional
+                posterior.
+            condition: Parameter set that all dimensions not specified in
+                `dims_to_sample` will be fixed to. Should contain dim_theta elements,
+                i.e. it could e.g. be a sample from the posterior distribution.
+                The entries at all `dims_to_sample` will be ignored.
+            dims_to_sample: Which dimensions to sample from. The dimensions not
+                specified in `dims_to_sample` will be fixed to values given in
+                `condition`.
+        """
+
+        self.potential_fn_provider = potential_fn_provider
+        self.condition = ensure_theta_batched(condition)
+        self.dims_to_sample = dims_to_sample
+
+    def __call__(self, prior, net: nn.Module, x: Tensor, mcmc_method: str,) -> Callable:
+        """Return potential function.
+
+        Switch on numpy or pyro potential function based on `mcmc_method`.
+        """
+        _ = self.potential_fn_provider.__call__(prior, net, x, mcmc_method)
```

Comment:
We could comment here: "Set prior, net, x as attributes of unconditional potential_fn_provider"